facia/app/controllers/FaciaController.scala

```diff
@@ -352,27 +353,60 @@ trait FaciaController
 
   def renderShowMore(path: String, collectionId: String): Action[AnyContent] =
     Action.async { implicit request =>
-      frontJsonFapi.get(path, fullRequestType).flatMap {
-        case Some(pressedPage) =>
-          val containers = Front.fromPressedPage(pressedPage, Edition(request), adFree = request.isAdFree).containers
-          val maybeResponse =
-            for {
-              (container, index) <- containers.zipWithIndex.find(_._1.dataId == collectionId)
-              containerLayout <- container.containerLayout
-            } yield {
-              val remainingCards: Seq[FaciaCardAndIndex] = containerLayout.remainingCards.map(_.withFromShowMore)
-              val adFreeFilteredCards: Seq[FaciaCardAndIndex] = if (request.isAdFree) {
-                remainingCards.filter(c => !checkIfPaid(c.item))
-              } else {
-                remainingCards
+      if (request.forceDCR) {
+        frontJsonFapi.get(path, fullDCRRequestType).flatMap {
+          case Some(pressedPage) =>
+            // TODO: Do we need to add anything else?
+
+            // TODO: The issue here is that .head will throw an error if the collection isn't found
+            //  I can't work out how to extend the 'for' structure below to this, though.
+            //  If the collection isn't found, will we need a new 'path', or will one of the other return
+            //  values cover this already?
+            val matchedCollection = pressedPage.collections.filter(_.id == collectionId).head
```

Comment:
@jamesgorrie as mentioned in the inline comments above, this `.head` call can throw an error if the expected `collectionId` isn't found, so we need to restructure.

As hinted at in the inline comments ln.379-381, I think the choice we make for handling this case will also shape how we should handle errors downstream (e.g. from failure of the POST request to DCR).